### ISSUE TITLE
Game Crashes when deleting or rebasing long flight plan after last maneuver execution

### ISSUE THREAD

**Issue by fat-lobyte** [mockingbirdnest/Principia#2953]
I can make the game crash almost reliably by pressing on the "Delete Flight" or "Rebase" button:
![screenshot97](https://user-images.githubusercontent.com/173609/115307894-2ab45a00-a16a-11eb-8183-5289d1db2fcd.png)

This is after executing the planned maneuver burn.

Workaround is to reduce the flight plan length first

Quicksave is available on request, but it's a full RP-1 install and around 70mb.

Logs:
[INFO.20210419-221504.4588.log](https://github.com/mockingbirdnest/Principia/files/6339257/INFO.20210419-221504.4588.log)
[WARNING.20210419-221504.4588.log](https://github.com/mockingbirdnest/Principia/files/6339258/WARNING.20210419-221504.4588.log)
[ERROR.20210419-221504.4588.log](https://github.com/mockingbirdnest/Principia/files/6339259/ERROR.20210419-221504.4588.log)

This is from the Windows Event log, sorry for the German:
```
Windows hat diagnostiziert, dass der virtuelle Speicher unzureichend ist. Die folgenden Programme belegten den meisten virtuellen Speicher: KSP_x64.exe (4588) belegt 18047619072 Bytes, qbittorrent.exe (7692) belegt 553623552 Bytes und firefox.exe (11520) belegt 253833216 Bytes.
```

**Comment by pleroy** [comment]
As I am sure you noticed from the logs, there is no crash of Principia *per se*.  This doesn't mean that there is no problem, or that Principia is not the cause, but surely there is no smoking gun.  If we do run out of memory, we are going to die in mysterious ways.

I don't think that the save would tell us much.  Would you by any chance be able to give us a [journal](https://github.com/mockingbirdnest/Principia/wiki/Installing,-reporting-bugs,-and-frequently-asked-questions#journaling)?  That's the only way that we can hope to reproduce the problem.  (Even so, it's not a done deal: my system may or may not run out of memory in the same circumstances as yours does.)

**Comment by fat-lobyte** [comment]
This time with a shorter flight plan it did not crash, but I did have a long freeze.

Here's the journal and logs: https://drive.google.com/file/d/1FW_iPiqm41QR0704kCj17pa4eyjdH1fB/view?usp=sharing

Looks like a duplicate of #2957

**Comment by pleroy** [comment]
With Grassmann, this journal has a `FlightPlanDelete` which takes 73.9 s on my machine.  With #2959, this operation drops below 100 ms.  So this is indeed the same problem as #2957.

**Comment by pleroy** [comment]
Fixed by #2959.
